develop/develop-images/build_enhancements.md

```diff
@@ -126,48 +126,60 @@ checks for updates of the syntax before building, making sure you are using the
 most current version. Learn more about the `syntax` directive in the
 [Dockerfile reference](/engine/reference/builder/#syntax).
 
-## New Docker Build secret information
+## Securely managing secrets during Docker Build
 
-The new `--secret` flag for docker build allows the user to pass secret
-information to be used in the Dockerfile for building docker images in a safe
-way that will not end up stored in the final image.
+`docker build` understands the `--secret` flag to allow the user to pass secret
+information that can be referenced in the Dockerfile.  The secret values are
+only used for building docker images and will not end up stored in the final
+image.
 
-`id` is the identifier to pass into the `docker build --secret`. This identifier
-is  associated with the `RUN --mount` identifier to use in the Dockerfile. Docker
-does not use the filename of where the secret is kept outside of the Dockerfile,
-since this may be sensitive information.
+A build operation may have multiple secrets associated with it, each identified
+with an `id` which is shared between the `docker build` command line flag and
+the Dockerfile.
 
-`dst` renames the secret file to a specific file in the Dockerfile `RUN` command
-to use.
+### Accessing Secrets during the build
 
-For example, with a secret piece of information stored in a text file:
-
-```console
-$ echo 'WARMACHINEROX' > mysecret.txt
-```
-
-And with a Dockerfile that specifies use of a BuildKit frontend
-`docker/dockerfile:1.2`, the secret can be accessed when performing a `RUN`:
+Secrets are accessed in Dockerfiles with a `RUN --mount=type=secret,id=<id>`
+command.  The value of the secret with the given `id` will be mounted as a file
+in the file-system for the duration of that `RUN` command and will otherwise be
+invisible to the rest of the build and to uses of the image.
 
 ```dockerfile
 # syntax=docker/dockerfile:1.2
 
 FROM alpine
 
-# shows secret from default secret location:
+# Fetch secret from default secret location:
 RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
-
-# shows secret from custom secret location:
-RUN --mount=type=secret,id=mysecret,dst=/foobar cat /foobar
 ```
 
-The secret needs to be passed to the build using the `--secret` flag.
-This Dockerfile is only to demonstrate that the secret can be accessed. As you
+_This Dockerfile is only to demonstrate that the secret can be accessed. As you
 can see the secret printed in the build output. The final image built will not
-have the secret file:
+have the secret file._
+
+The `type=secret` and `id` parameters are mandatory but the following optional
+parameters are also available if desired:
```

Comment:
This is not a complete list https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#run---mounttypesecret